services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/QinQFlowSpec.groovy

```diff
@@ -0,0 +1,365 @@
+
+        then: "Response contains correct info about innerVlanIds"
+        with(response) {
+            it.source.vlanId == (srcVlanId ? srcVlanId : srcInnerVlanId)
+            it.source.innerVlanId == (srcVlanId ? srcInnerVlanId : 0)
+            it.destination.vlanId == (dstVlanId ? dstVlanId : dstInnerVlanId)
+            it.destination.innerVlanId == (dstVlanId ? dstInnerVlanId : 0)
+        }
+
+        and: "Flow is really created with requested innerVlanIds"
+        with(northbound.getFlow(flow.flowId)) {
+            it.source.vlanId == (srcVlanId ? srcVlanId : srcInnerVlanId)
+            it.source.innerVlanId == (srcVlanId ? srcInnerVlanId : 0)
+            it.destination.vlanId == (dstVlanId ? dstVlanId : dstInnerVlanId)
+            it.destination.innerVlanId == (dstVlanId ? dstInnerVlanId : 0)
+        }
+
+        //is not implemented yet
+//        and: "Flow is valid and pingable"
+//        northbound.validateFlow(flow.flowId).each { assert it.asExpected }
+//        verifyAll(northbound.pingFlow(flow.flowId, new PingInput())) {
+//            it.forward.pingSuccess
+//            it.reverse.pingSuccess
+//        }
+
+        //is not implemented yet
+//        and: "Involved switches pass switch validation"
+//        with(pathHelper.getInvolvedSwitches(pathHelper.convert(northbound.getFlowPath(flow.flowId)))) {
+//            def validationInfo = northbound.validateSwitch(it.dpId)
+//            validationInfo.verifyRuleSectionsAreEmpty(["missing", "excess", "misconfigured"])
+//            validationInfo.verifyMeterSectionsAreEmpty(["missing", "excess", "misconfigured"])
+//        }
+
+        when: "Delete the flow"
+        flowHelperV2.deleteFlow(flow.flowId)
+
+        then: "Flow rules are deleted"
+        //is not implemented yet
+//        Wrappers.wait(RULES_INSTALLATION_TIME, 1) {
+//            assert northbound.getSwitchRules(sw.dpId).flowEntries*.cookie.sort() == sw.defaultCookies.sort()
+//        }
+
+        where:
+        srcVlanId | srcInnerVlanId | dstVlanId | dstInnerVlanId
+        10        | 20             | 30        | 40
+        10        | 20             | 30        | 0
+        10        | 20             | 0         | 0
+        10        | 0              | 0         | 40
+        0         | 20             | 30        | 0
+    }
+
+    @Tags(TOPOLOGY_DEPENDENT)
+    def "System doesn't allow to create a QinQ flow when a switch doesn't support multi table mode"() {
+        given: "A switch pair with disabled multi table mode on the src switch"
+        def swP = topologyHelper.getAllNeighboringSwitchPairs().find {
+            [it.src, it.dst].any { northbound.getSwitchProperties(it.dpId).multiTable }
+        } ?: assumeTrue("No suiting switches found", false)
+        def initSrcSwProps = northbound.getSwitchProperties(swP.src.dpId)
+        northbound.updateSwitchProperties(swP.src.dpId,
+                northbound.getSwitchProperties(swP.src.dpId).tap { multiTable = false })
```

Comment:
you can use `initSrcSwProps.jacksonCopy().tap {...}`

Reply:
I will do it when original PR is merged/rebased

Reply:
fixed